src/processors/knex-processor.ts

```diff
@@ -245,24 +249,32 @@ export default class KnexProcessor<
     const relationProcessor: KnexProcessor<Resource> = (await this.processorFor(
       relationship.type().type
     )) as KnexProcessor<Resource>;
+
     const query = relationProcessor.getQuery();
     const foreignTableName = relationProcessor.tableName;
     const sqlOperator = Array.isArray(result) ? "in" : "=";
+
+    const primaryKey = this.resourceClass.schema.primaryKeyName || "id";
+
     const queryIn: string | string[] = Array.isArray(result)
-      ? result.map((a: Resource) => a.id)
-      : result.id;
+      ? result.map((a: Resource) => a[primaryKey])
```

Comment:
@spersico can we name this variable `resource`? `a` means nothing in this context...